components/camel-kudu/src/main/java/org/apache/camel/component/kudu/KuduEndpoint.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.kudu;
+
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.kudu.client.KuduClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents a Kudu endpoint. A kudu endpoint allows you to interact with
+ * <a href="https://kudu.apache.org/">Apache Kudu</a>, a free and open source
+ * column-oriented data store of the Apache Hadoop ecosystem.
+ */
+@UriEndpoint(firstVersion = "2.25.0",
+    scheme = "kudu",
+    title = "Apache Kudu", syntax = "kudu:type",
+    consumerClass = KuduConsumer.class,
+    label = "database,iot")
+public class KuduEndpoint extends DefaultEndpoint {
```

Comment:
@Delawen may be rethink about enpoint syntax from separate URI parameters (host and port) to the syntax like in thrift or similar components, i.e. thrift:host:port/service.
for this component it would like kudu:host:port/type that much more cloud ready for services and routes.

Reply:
Makes sense. Thanks!